js_modules/dagit/packages/ui/src/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+
+import {Colors} from './Colors';
+
+export type ErrorCollectionContextValue = {
+  errorStackIncluded: boolean;
+  errorCollectionMessage: string;
+  onReportError: (error: Error, context: Record<string, any>) => void;
+};
+
+export const ErrorCollectionContext = React.createContext<ErrorCollectionContextValue>({
+  errorStackIncluded: true,
+  errorCollectionMessage:
+    `Please report this error to the Elementl team via GitHub or Slack. ` +
+    `Refresh the page to try again.`,
+  onReportError: (error, context) => {
+    console.error(error, context);
+  },
+});
+
+interface ErrorBoundaryProps {
+  region: string;
+  resetErrorOnChange?: any[];
+}
+
+interface ErrorBoundaryState {
+  error: Error | null;
+  errorResetPropsValue: string | null;
+}
+
+export class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  state: ErrorBoundaryState = {error: null, errorResetPropsValue: null};
+
+  static contextType = ErrorCollectionContext;
+
+  componentDidUpdate() {
+    if (
+      this.state.error &&
+      this.state.errorResetPropsValue !== JSON.stringify(this.props.resetErrorOnChange)
+    ) {
+      this.setState({error: null, errorResetPropsValue: null});
+    }
+  }
+
+  componentDidCatch(error: Error, info: any) {
+    // You can also log the error to an error reporting service
+    // Note: Sentry may be blocked by an ad-blocker, can't assume it's available
+    this.context.onReportError(error, {info, region: this.props.region});
+    this.setState({error, errorResetPropsValue: JSON.stringify(this.props.resetErrorOnChange)});
+  }
+
+  render() {
+    const {error} = this.state;
+    const {errorCollectionMessage, errorStackIncluded} = this
+      .context as ErrorCollectionContextValue;
+
+    if (error) {
+      return (
+        <div
+          style={{
+            width: '100%',
+            height: '100%',
+            flex: 1,
+            border: `1px solid ${Colors.HighlightRed}`,
+            background: 'white',
+            borderRadius: 3,
+            color: '#222',
+            padding: 15,
+          }}
+        >
+          <h3>Sorry, {this.props.region} can&apos;t be displayed.</h3>
+          <p style={{color: Colors.Gray300}}>{errorCollectionMessage}</p>
+          {errorStackIncluded && (
+            <code>
+              <pre>{`${error}`}</pre>
+            </code>
```

Comment:
Style the `<pre>` instead of wrapping it in `<code>`, since `<code>` is an inline element?